src/cryptonote_core/blockchain.cpp

```diff
@@ -2530,7 +2530,7 @@ bool Blockchain::check_tx_inputs(transaction& tx, tx_verification_context &tvc,
       }
     }
 
-    if (hf_version >= HF_VERSION_MIN_MIXIN_10 && mixin != 10)
+    if (((hf_version == HF_VERSION_MIN_MIXIN_10 || hf_version == HF_VERSION_MIN_MIXIN_10+1) && mixin != 10) || (hf_version >= HF_VERSION_MIN_MIXIN_10+2 && mixin > 10))
```

Comment:
That's correct. The first version rejects ring size eg 10 in version 8 and 9. The new version will accept it. That's what the commit message says, unless I'm missing something.
The code just after it will reject small ring sizes if the special case does not apply.

Reply:
The new version seems to accept any ring sizes equal to or less than 11, _regardless of whether outputs are mixable or unmixable_. Isn't this very wrong?

Reply:
Sorry, I misread. The code looks correct.